### PR TITLE
[FEATURE] Support new "cname" option to deliver resource with privatecdn

### PR DESCRIPTION
**Pull request by dhuf**
Backport from v2

### CODE REVIEW

File: Resources/Private/Language/backend.xlf [hunk 1]
```diff
@@ -14,10 +14,13 @@
                 <source>Cloudinary API Secret</source>
             </trans-unit>
             <trans-unit id="driverConfiguration.basePath">
-                <source>A possible base path to store files in a parent directory. Example `production/`</source>
+                <source>A possible base path for storing files in a parent directory on cloudianry. Example `production/`</source>
```
Comment (SomeBdyElse) [inline Resources/Private/Language/backend.xlf]:
```suggestion
                <source>A possible base path for storing files in a parent directory on cloudinary. Example `production/`</source>
```

File: Classes/Services/ConfigurationService.php [hunk 2]
```diff
@@ -33,8 +27,8 @@ public function __construct(array $configuration)
      */
     public function get(string $key): string
     {
-        $value = (string)$this->configuration[$key];
-        if (preg_match('/^%(.*)%$/', $value, $matches)) {
+        $value = trim((string)$this->configuration[$key]);
+        if (preg_match('/^%\w+\((.*)\)%$/', $value, $matches) || preg_match('/^%(.*)%$/', $value, $matches)) {
```
Comment (SomeBdyElse) [inline Classes/Services/ConfigurationService.php]: `\w+` will match more than just `env`. Why not limit to "env\(" here?

File: Classes/Utility/CloudinaryApiUtility.php [hunk 3]
```diff
@@ -31,7 +31,9 @@ public static function initializeByConfiguration(array $configuration)
                 'api_key' => $configurationService->get('apiKey'),
                 'api_secret' => $configurationService->get('apiSecret'),
                 'timeout' => $configurationService->get('timeout'),
-                'secure' => true
+                'secure' => true,
+                'private_cdn' => $configurationService->get('cname') !== '',
```
Comment (SomeBdyElse) [inline Classes/Utility/CloudinaryApiUtility.php:35]: $configurationService->get('cname') will throw an exception if `cname` is set to `env(…)` but the env does not exist. Is this intended?

File: Classes/Services/ConfigurationService.php (hunk 2 above)
Comment (SomeBdyElse) [reply Classes/Services/ConfigurationService.php]:
```suggestion
        if (preg_match('/^%env\((.*)\)%$/', $value, $matches) || preg_match('/^%(.*)%$/', $value, $matches)) {
```

File: Classes/Utility/CloudinaryApiUtility.php (hunk 3 above)
Comment (jrenggli) [reply Classes/Utility/CloudinaryApiUtility.php:35]: IMHO this is intended. As a reference to a non-existent environment variable is a clear configuration error.

The exception thrown

```php
throw new \RuntimeException(sprintf('No value found for environment variable "%s"', $matches[1]), 1626948978);
```

helps to find and fix this error.

**Pull request merged**